src/firehose.rs

```diff
@@ -0,0 +1,3 @@
+//! Amazon Kinesis Firehose
+
```

Comment:
The `FooException` naming scheme is actually Amazon's, not mine.  i.e., those names come right out of the botocore definitions.  We could do a string substitution to name them `FooError` or something more idiomatic I suppose.